lib/modules/database/reference.js

```diff
@@ -157,19 +157,25 @@ export default class Reference extends ReferenceBase {
    * @param onComplete
    * @param applyLocally
    */
-  transaction(transactionUpdate, onComplete?: () => any, applyLocally: boolean = false) {
+  transaction(
+    transactionUpdate: Function,
+    onComplete?: (?Error, any, ?Snapshot) => any,
+    applyLocally: boolean = false
+  ) {
     if (!isFunction(transactionUpdate)) return Promise.reject(new Error('Missing transactionUpdate function argument.'));
 
     return new Promise((resolve, reject) => {
       const onCompleteWrapper = (error, committed, snapshotData) => {
         if (error) {
-          if (isFunction(onComplete)) onComplete(error, committed, null);
+          if (typeof onComplete === 'function') {
+            onComplete(error, committed, null);
+          }
           return reject(error);
         }
 
         const snapshot = new Snapshot(this, snapshotData);
 
-        if (isFunction(onComplete)) {
+        if (typeof onComplete === 'function') {
```

Comment:
Not nit-picking here but any reason you changed this one? There's one above also on 165